### ISSUE TITLE
Misleading doc string for callFromThread

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2449]
|<img alt="Scramblejams's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Scramblejams reported|
|-|-|
|Trac ID|trac#2449|
|Type|defect|
|Created|2007-02-12 17:17:01Z|

In...

http://twistedmatrix.com/documents/current/api/twisted.internet.interfaces.IReactorThreads.html

...the docs for reactor.callFromThread() state that the callable must be threadsafe. But according to Eric Mangold...

http://twistedmatrix.com/pipermail/twisted-python/2007-February/014855.html

...this is wrong.

Attachments:

* [misleading-callFromThread-doc.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/432/4323bf77eeca0bbf71ee86d5531a2fe1cb4e21c3/069f6e1dc46e47b408d0db5cdad0647a96ad836d.patch) (1316 bytes) - added by thijs on 2008-11-09 17:20:21Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__2449 2449
type__defect defect
reporter__Scramblejams Scramblejams
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation__threads documentation, threads
time__1171300621000000 1171300621000000
changetime__1226844691000000 1226844691000000
version__None None
owner__ 
cc__exarkun cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

Coming up with a phrasing to describe the behavior which is more clear and concise than the current wording is tricky.  Do you have any suggestions?

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Replying to [exarkun](#note_2):
> Coming up with a phrasing to describe the behavior which is more clear and concise than the current wording is tricky.  Do you have any suggestions?

In that case let's use that alternative which is better than the current misleading doc string. See attached patch.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set owner to @thijstriemstra|
|-|-|

Please apply.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [25371]) Apply misleading-callFromThread-doc.patch: make docs for reactor.callFromThread() clear about when the callable must be threadsafe.

Author: thijs
Reviewer: therve
Fixes: [#2449](https://github.com/twisted/twisted/issues/2449)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
